### DISCUSSION TITLE
[SOLVED] New installation, login view invalid

### DISCUSSION THREAD

**Discussion by mauriceb77** [codeigniter4/shield#456] [Q&A] [answered]
I just installed shield on my existing CI4 app, I used composer and spark and all files look OK. I setup "session" in my global filter as I want the entire site protected but I am receiving the following error. I tried a reinstall to overwrite the config files as this was a solution in CI forum but it did not resolve the issue.

Invalid file: \CodeIgniter\Shield\Views\login.php
SYSTEMPATH\Exceptions\FrameworkException.php at line 33

Global config:
```php
    public $globals = [
        'before' => [
            'csrf',
            'session' => ['except' => ['login*', 'register*']],
        ],
        'after' => [
            'toolbar',
            // 'honeypot',
        ],
    ];
```

**Comment by kenjis** [comment, marked as answer]
Can you show the all back trace?
Do you have `vendor/codeigniter4/shield/src/Views/login.php` ?

**Reply by mauriceb77** [reply]
Yes, see attached.
![Capture](https://user-images.githubusercontent.com/103624677/192396779-b5bcc54b-2146-44a8-95f2-a011480e3b9f.PNG)

**Reply by MGatner** [reply]
Maybe a Windows path issue. Could you share the entire backtrace? That would help us pinpoint the source. 

Also did you use the `shield:setup` command? Or are you using it just as an external library?

**Reply by mauriceb77** [reply]
Full trace attached. I used composer require codeigniter4/shield:dev-develop and php spark shield:setup.
[CodeIgniter_View_Exceptions_ViewException.pdf](https://github.com/codeigniter4/shield/files/9656604/CodeIgniter_View_Exceptions_ViewException.pdf)

**Reply by datamweb** [reply]
Hello @mauriceb77 
Your CI version is 4.1.4.
Shield requires **Codeigniter v4.2.3 or later**.
Please update the CI framework.
https://codeigniter.com/user_guide/installation/upgrading.html

**Reply by mauriceb77** [reply]
Thank you. I just learned of shield and overlooked the version requirement.

**Reply by datamweb** [reply]
No problem, welcome!

**Reply by mauriceb77** [reply]
One more question, I am upgrading using composer and it looks like I just need to use "> composer update" after each version, I am now on 4.1.4 but the update is not incrementing, I now receive message below.

>Loading composer repositories with package information
Updating dependencies
Nothing to modify in lock file
Installing dependencies from lock file (including require-dev)
Nothing to install, update or remove
Generating autoload files
61 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
> CodeIgniter\ComposerScripts::postUpdate

**Reply by datamweb** [reply]
Please read [here](https://codeigniter.com/user_guide/installation/upgrading.html), if your problem is not solved,
Register your question in [CodeIgniter 4 Support](https://forum.codeigniter.com/forumdisplay.php?fid=30).
This discussion is related to **Shield**, your question is related to **CodeIgniter 4** .
